keepalived/vrrp/vrrp_print.c

```diff
@@ -418,3 +421,182 @@ vrrp_print_stats(void)
 	}
 	fclose(file);
 }
+
+#ifdef _JSON_
+void
+vrrp_print_json(void)
+{
+	FILE *file;
+	file = fopen ("/tmp/keepalived.json","w");
+	if (!file) {
+		log_message(LOG_INFO, "Can't open /tmp/keepalived.json (%d: %s)",
+			errno, strerror(errno));
+		return;
+	}
+
+	list l = vrrp_data->vrrp;
+	if LIST_ISEMPTY(vrrp_data->vrrp) {
+		log_message(LOG_INFO, "vrrp_data->vrrp is empty");
+		return;
+	}
+
+	element e, f;
+	vrrp_t *vrrp;
+	struct json_object *array, *vips;
+	array = json_object_new_array();
+
+	for (e = LIST_HEAD(l); e; ELEMENT_NEXT(e)) {
+		vrrp = ELEMENT_DATA(e);
+		list v = vrrp->vip;
+		struct json_object *instance_json, *json_stats, *json_data;
```

Comment:
There is inconsistency about where some of the variables that are only used in the loop are declared. Either they should all be declared at the beginning of the function, or all at the start of the loop.